nolan-swerve-2023/src/main/java/frc/robot/subsystems/SwerveDriveSubsystem.java

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.sensors.WPI_Pigeon2;
+import com.pathplanner.lib.PathPlanner;
+import com.pathplanner.lib.PathPlannerTrajectory;
+import com.pathplanner.lib.PathPlannerTrajectory.PathPlannerState;
+import com.pathplanner.lib.commands.PPSwerveControllerCommand;
+import com.swervedrivespecialties.swervelib.SwerveModule;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveDriveOdometry;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class SwerveDriveSubsystem extends SubsystemBase {
+	private static SwerveDriveSubsystem instance = null;
+
+	public static SwerveDriveSubsystem getInstance() {
+		return instance;
+	}
+
+	private SwerveDriveKinematics m_kinematics;
+
+	private WPI_Pigeon2 m_pigeon;
+
+	private SwerveModule m_frontLeftModule;
+	private SwerveModule m_frontRightModule;
+	private SwerveModule m_backLeftModule;
+	private SwerveModule m_backRightModule;
+
+	private ChassisSpeeds m_chassisSpeeds = new ChassisSpeeds(0.0, 0.0, 0.0);
+
+	private Configuration m_config;
+
+	private SwerveDriveOdometry m_odometry;
+
+	public static class Configuration {
```

Comment:
Fix these configs to represent what they actually are and include units. I was bad, and just left them as I copied them.